File: library/src/main/java/me/proxer/library/api/forum/ForumApi.java
package me.proxer.library.api.forum;

import retrofit2.Retrofit;

/**
 * API for the Forum class.
 *
 * @author Ruben Gees
 */
public class ForumApi {

    private final InternalApi internalApi;

    /**
     * Only for internal use.
     */
    public ForumApi(final Retrofit retrofit) {
        this.internalApi = retrofit.create(InternalApi.class);
    }

    /**
     * Returns the respective endpoint.
     */
    public TopicEndpoint topic(final String id) {
        return new TopicEndpoint(internalApi, id);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/forum/ForumApi.kt
package me.proxer.library.api.forum

import retrofit2.Retrofit

/**
 * API for the Forum class.
 *
 * @author Ruben Gees
 */
class ForumApi internal constructor(retrofit: Retrofit) {

    private val internalApi: InternalApi = retrofit.create(InternalApi::class.java)

    /**
     * Returns the respective endpoint.
     */
    fun topic(id: String) = TopicEndpoint(internalApi, id)
}
